ykcapi/src/lib.rs

```diff
@@ -95,29 +100,66 @@ impl Registers {
     }
 }
 
+#[derive(Debug)]
+#[repr(C)]
+struct AOTVar {
+    bbidx: u32,
+    instridx: u32,
+    fname: *const i8,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct AOTMap {
+    addr: *const c_void,
+    size: usize,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct CurPos {
+    bbidx: u32,
+    instridx: u32,
+    fname: *const i8,
+}
+
 /// Parses the stackmap and saved registers from the given address (i.e. the return address of the
 /// deoptimisation call).
-/// FIXME: Until we have the stopgap interpreter we simply print out the values we find.
 #[cfg(target_arch = "x86_64")]
 #[no_mangle]
 pub extern "C" fn yk_stopgap(
     sm_addr: *const c_void,
     sm_size: usize,
+    aotmap: &AOTMap,
+    curpos: &CurPos,
     retaddr: usize,
     rsp: *const c_void,
 ) {
     // FIXME: remove once we have a stopgap interpreter.
     eprintln!("jit-state: stopgap");
+
+    // Parse AOTMap.
+    let aotmap = unsafe { slice::from_raw_parts(aotmap.addr as *const AOTVar, aotmap.size) };
+
     // Restore saved registers from the stack.
     let registers = Registers::from_ptr(rsp);
 
+    let mut sginterp = unsafe {
+        SGInterp::new(
+            curpos.bbidx,
+            curpos.instridx,
+            std::ffi::CStr::from_ptr(curpos.fname),
+        )
+    };
+
     // Parse the stackmap.
     let slice = unsafe { slice::from_raw_parts(sm_addr as *mut u8, sm_size) };
     let map = StackMapParser::parse(slice).unwrap();
     let locs = map.get(&retaddr.try_into().unwrap()).unwrap();
 
     // Extract live values from the stackmap.
-    for l in locs {
+    // Skip first 3 locations as they don't relate to any of our live variables.
```

Comment:
This is too dangerous to be an in-code constant: it must be a named constant so that we can search for it later if/when it needs to be changed.